### PR TITLE
NW4- Maha Malik - JS Core1 - Week 3

### PR DESCRIPTION
**Pull request by maham511**
# Your Details

- Your Name: Maha Malik
- Your City: NW4
- Your Slack Name: Maha Malik

# Homework Details

- Module: JS C1
- Week: 3

# Notes

- What did you find easy?
Using conditional operators to set conditions, 

- What did you find hard?
Most exercises and mandatory tasks felt too advanced at this stage, took too long to figure what methods to use, how problems could be approached as knowledge is very limited.

- What do you still not understand?
 Same as above, also calling functions within other functions, loops

- Any other notes?
Feel like we still need a lot more practice with simpler JS problems. This week's tasks could have been lower in either quantity or difficulty level.

### CODE REVIEW

**Review by Texx45 — COMMENTED** [review]
Nice work Maha! I really need to put more comments in my code as I can clearly see what your code is doing straight away. Sometimes I have to work mine out!

**Comment by maham511** [comment]
Thanks Terry, I'm the same with mine without comments. Really does help!

File: 1-exercises/E-array-map/exercise.js [hunk 1]
```diff
@@ -2,3 +2,34 @@
...
+});
+
+//curly bracket and return keyword not needed either
+var map5 = numbers.map(number => number * 100);
+
+console.log(map1);
+console.log(map2);
+console.log(map3);
+console.log(map4);
+console.log(map5);
```
Comment (lannem) [inline 1-exercises/E-array-map/exercise.js:35]: very nice, well done on doing 5 different versions!

File: 1-exercises/F-array-forEach/exercise.js [hunk 2]
```diff
@@ -9,6 +9,34 @@
...
+       
+    }
+  
+    let numToWord = arr.map(changeToWord); //prints relevant words instead of multiples
+
+  //Combine map with forEach
+  numToWord.forEach(function (number) {
+    console.log(number);
+  });  
+
```
Comment (lannem) [inline 1-exercises/F-array-forEach/exercise.js:39]: nice 👍

File: 1-exercises/H-array-methods-2/exercise.js [hunk 3]
```diff
@@ -15,8 +15,8 @@ var everyone = [
   "Swathi",
 ];
 
-var firstFive; // complete this statement
-var lastFive; // complete this statement
+var firstFive = everyone.slice(0,5); // complete this statement
+var lastFive = everyone.slice(2, 7); // complete this statement
```
Comment (lannem) [inline 1-exercises/H-array-methods-2/exercise.js:19]: you could also have done `.slice(-5)` here :)

File: 2-mandatory/1-create-functions.js [hunk 4]
```diff
@@ -24,16 +29,31 @@ Write a function that:
...
 
-function remove() {
+function remove(arr, index) {
+  //Psuedo
+  // take array and index as parameter
+  //create new array
+  //new array has same elements, minus indexed element
+
+  const arrayCopy = [...arr];   //Creates copy of original array
+  arrayCopy.splice(index, 1);   //Removes element from arrayCopy provided in index parameter, 1 tells splice to only remove needed element.
```
Comment (lannem) [inline 2-mandatory/1-create-functions.js:55]: nice use of splice!

File: 2-mandatory/6-journey-planner.js [hunk 5]
```diff
@@ -121,10 +132,20 @@ function getLocationName() {}
    
   Advanced challange: try to use arrow function when invoking an array method.
 */
+
+//main function doesn't work, didn't manage to solve
 function journeyPlanner(locations, transportMode) {
   // Implement the function body
+      if (locations.map(isAccessibleByTransportMode())) {
```
Comment (lannem) [inline 2-mandatory/6-journey-planner.js:139]: you need to pass in parameters to `isAccessibleByTransportMode`

File: 2-mandatory/8-password-validator.js [hunk 6]
```diff
@@ -23,8 +23,20 @@ Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
 
 */
+//main function doesn't work, didn't manage to solve
+  function validatePasswords(string) {
 
-function validatePasswords(passwords) {}
+      containsLowercaseLetter() &&
```
Comment (lannem) [inline 2-mandatory/8-password-validator.js:29]: you need to pass in the `string` parameter to these functions as it contains the password :)

**Review by lannem — COMMENTED** [review]
well done Maha, you had a couple of failing tests but they only require small changes to get them working :) let me know if you need any help

**Comment by maham511** [comment]
Thanks Lucy for the feedback, will try an get the failing tests sorted. :)

Regards,
Maha

________________________________

**Pull request closed without merging**